lib/hal/video.c

```diff
@@ -227,6 +231,17 @@ BOOLEAN XVideoListModes(VIDEO_MODE *vm, int bpp, int refresh, void **p)
 		if(pVidMode->refresh != refresh)
 			continue;
 
+		if(eepromVidSize & 0x000E0000)
+		{
+			if(pVidMode->height == 480 && !(eepromVidSize & 0x00080000))
+				continue;
+
+			if(pVidMode->height == 720 && !(eepromVidSize & 0x00020000))
+				continue;
+
+			if(pVidMode->height == 1080 && !(eepromVidSize & 0x00040000))
+				continue;
```

Comment:
This code is very non-descriptive. Introduce constants for the masks or at least move result into temporaries: `allow_480p = [...]` / `allow_720p = [...]` / `allow_1080p = eepromVidSize & 0x00040000;` etc.

Reply:
Better now?